### PR TITLE
3d plot view angle documentation

### PR DESCRIPTION
**Pull request by scottshambaugh**
## PR Summary

Adds documentation and images for the 3d plot view angle conventions. Also adds an example showing the primary view planes. Closes https://github.com/matplotlib/matplotlib/issues/21508

Inkscape file to generate the view angles plot is attached (the arrows render weird in the svg preview here for some reason).
mplot3d_view_axes

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by scottshambaugh** [comment]
New page in the docs: https://output.circle-artifacts.com/output/job/69e56563-d95d-4437-95d2-7934335bbf8d/artifacts/0/doc/build/html/api/toolkits/mplot3d/view_angles.html

**Comment by oscargus** [comment]
I think that it is possible to store an svg-file and use that? (There's been a similar case in SymPy reducing the file size from MBs to kb... I do not know if any specific settings are required. But most browsers can render svg anyway?)

But the image looks good!

**Comment by timhoffm** [comment]
PNG size can be reduced by factors if put through a compressor like https://tinypng.com/. I think if applied to both images, sizes are reasonable small.

Content:
1. The view direction line changes from green to black, which looks weird to me:
grafik

2. I find the view plane a bit hard to understand, though after thinking about it, I believe it's correct.
Would it be helpful to add the camera position and view cone similar to
grafik
(don't need the fancy camera picture necessarily :))

**Comment by scottshambaugh** [comment]
PNGs are smaller, thanks for the site. The issue with svg is that different browsers render differently, and we actually have a great example right here. In the screenshot of the svg image for @timhoffm it looks like the arrows are not present for some reason, whereas I see arrows colored black in the previews, and when I open in inkscape the arrows colored the same as their lines (what I intended and what the png shows). Tim, do the arrows there resolve the overlaying lines looking weird?
image

Updated to add a little camera cone, I'm back and forth on if it adds more clutter than it helps. Thoughts?

**Comment by story645** [comment]
Wondering if for this PR or a follow on, the section in the 3D FAQ can be added/consolidated into the doc?

**Comment by scottshambaugh** [comment]
Are you asking for the view angles to be added to the FAQ, or the FAQ info to be interspersed elsewhere where it makes sense?

**Comment by story645** [comment]
The relevant parts of the FAQ can be incorporated into the new document so that information isn't spread across two.

**Comment by scottshambaugh** [comment]
I don't see anything in the FAQ that I think it makes sense to pull out to the new page, but I'm happy to just put all the new info into the existing FAQ.

**Comment by story645** [comment]
Then just forget my suggestion since I like this is a standalone & think it's more discoverable that way (I'm not a big fan of FAQs)

**Comment by scottshambaugh** [comment]
Decided to test the claim that you could print out and fold the primary planes image into a box:

0B376F7A-503B-4BB0-B373-B731B3837453

**Comment by timhoffm** [comment]
The camera looks good and the arrow also resolves my line issue 👍.

One additional idea for decluttering: How about only indicating the direction roll direction:

image

Technically, that's a bit different than the azimuth/elevation arrows, which show magnitude, but I think people still understand what roll is.

**Comment by oscargus** [comment]
OK! I get the size and rendering issue, but wouldn't it make sense to have the "box" as svg so that it can easily be resized when printing? (Given that the size is not completely off. Or generate that image on the fly? I assume it is generate with mpl?)

**Comment by scottshambaugh** [comment]
Ok, addressed those comments. I didn't totally go with Tim's suggestion since I think showing the roll angle zero reference is important, but I did take out two of the other lines there that weren't adding much. And to Oscar's suggestion, I made the unfolded box plot a gallery example with the generating code right there. I think it all looks good!

Built docs API page: https://output.circle-artifacts.com/output/job/0162ce41-cee3-47e2-a5a8-61aa17145914/artifacts/0/doc/build/html/api/toolkits/mplot3d/view_angles.html
Built docs gallery example: https://output.circle-artifacts.com/output/job/0162ce41-cee3-47e2-a5a8-61aa17145914/artifacts/0/doc/build/html/gallery/mplot3d/view_planes_3d.html

File: doc/api/toolkits/mplot3d/view_angles.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+mplot3d View Angles
+*******************
+
+How to define the view angle
+============================
+
+The position of the viewport "camera" in a 3D plot is defined by three angles:
+elevation, azimuth, and roll. From the resulting position, it always points
+towards the center of the plot box volume. The angle direction convention is
+shared with
```
Comment (timhoffm) [inline doc/api/toolkits/mplot3d/view_angles.rst:13]: Minor, but i think emphasizing the angle names makes sense.
```suggestion
*elevation*, *azimuth*, and *roll*. From the resulting position, it always
points towards the center of the plot box volume. The angle direction
convention is shared with
```

Also, could we say

"The angle direction is a common convention and e.g. shard with ..."

File: examples/mplot3d/view_planes_3d.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+         ('-XZ',   (0,  90, 0)),
+         ('-YZ',   (0, 180, 0))]
+
+layout = [['XY',  '.',   'L',   '.'],
+          ['XZ', 'YZ', '-XZ', '-YZ'],
+          ['.',   '.', '-XY',   '.']]
+fig, axd = plt.subplot_mosaic(layout, subplot_kw={'projection': '3d'},
+                              figsize=(12, 8.5))
+for i in range(len(axd) - 1):
+    plane = views[i][0]
```
Comment (timhoffm) [inline examples/mplot3d/view_planes_3d.py:33]: We don't need indices:
```suggestion
for plane, angles in views:
```

File: examples/mplot3d/view_planes_3d.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    axd[plane].set_xlabel('x')
+    axd[plane].set_ylabel('y')
+    axd[plane].set_zlabel('z')
+    axd[plane].set_proj_type('ortho')
+    axd[plane].view_init(elev=views[i][1][0],
+                         azim=views[i][1][1],
+                         roll=views[i][1][2])
```
Comment (timhoffm) [inline examples/mplot3d/view_planes_3d.py:40]:
```suggestion
    axd[plane].view_init(elev=angles[0],
                         azim=angles[1],
                         roll=angles[2])
```

File: examples/mplot3d/view_planes_3d.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    axd[plane].set_box_aspect(None, zoom=1.25)
+
+    label = f'{plane}\n{views[i][1]}'
```
Comment (timhoffm) [inline examples/mplot3d/view_planes_3d.py:43]:
```suggestion
    label = f'{plane}\n{angles}'
```

File: examples/mplot3d/view_planes_3d.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+for i in range(len(axd) - 1):
+    plane = views[i][0]
+    axd[plane].set_xlabel('x')
+    axd[plane].set_ylabel('y')
+    axd[plane].set_zlabel('z')
+    axd[plane].set_proj_type('ortho')
+    axd[plane].view_init(elev=views[i][1][0],
+                         azim=views[i][1][1],
+                         roll=views[i][1][2])
+    axd[plane].set_box_aspect(None, zoom=1.25)
```
Comment (timhoffm) [inline examples/mplot3d/view_planes_3d.py:41]: Would this be a good use for `axs[plane].axis('equal')`? 😄

**Review by timhoffm — COMMENTED** [review]
Looks very good.

Optional: It may be neat to add the fold lines to the unfolded view. But we can always add that later. I'd like to get this into 3.6 docs. So we should merge soon.

File: examples/mplot3d/view_planes_3d.py (hunk 5 above)
Comment (scottshambaugh) [reply examples/mplot3d/view_planes_3d.py:38]: A cutout should work for any aspect box :). But right now that command would only change the data limits! Need https://github.com/matplotlib/matplotlib/pull/23552 to merge in to change the box.

**Review by timhoffm — APPROVED** [review]

File: examples/mplot3d/view_planes_3d.py (hunk 5 above)
Comment (timhoffm) [reply examples/mplot3d/view_planes_3d.py:38]: Ok, let's to this later then. Since we've already released 3.6 RC1, #23552 can't go into 3.6 anymore.

**Comment by timhoffm** [comment]
@scottshambaugh thanks!

**Pull request merged**